R/DBConnection.R

```diff
@@ -106,6 +106,39 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbi_break_sentinel")
+
+dbFetchChunked <- function(rs, n, callback) {
+  while (TRUE) {
+    chunk <- dbFetch(rs, n = n)
+    if (nrow(chunk) == 0) {
+      return(chunk)
+    }
+    result <- callback(df = chunk)
+    if (identical(result, dbBreak)) {
+      return(chunk)
+    }
+  }
```

Comment:
What's a sensible return value here? Perhaps the results of all callbacks could be accumulated in a list, processed by a second callback, and then returned?

```
dbProcessChunked(..., callback, process_results, ...)
  while (...) {
    ...
    all_results <- c(all_results, list(callback(...)))
    ...
  }
  process_results(all_results)
}